crates/worker/src/partition/state_machine/tests/mod.rs

```diff
@@ -1016,3 +994,38 @@ async fn consecutive_exclusive_handler_invocations_will_use_inbox() -> TestResul
     test_env.shutdown().await;
     Ok(())
 }
+
+#[test(tokio::test)]
+async fn deduplicate_requests_with_same_pp_rpc_request_id() -> TestResult {
```

Comment:
Should we also check that the send the correct `SubmittedInvocationNotification` with the correctly set `is_new_invocation`?